pkg/kubectl/cmd/taint.go

```diff
@@ -0,0 +1,386 @@
+}
+
+// Complete adapts from the command line args and factory to the data required.
+func (o *TaintOptions) Complete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string) (err error) {
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	// retrieves resource and taint args from args
+	// also checks args to verify that all resources are specified before annotations
+	taintArgs := []string{}
+	metTaintArg := false
+	for _, s := range args {
+		isTaint := strings.Contains(s, "=") || strings.HasSuffix(s, "-")
+		switch {
+		case !metTaintArg && isTaint:
+			metTaintArg = true
+			fallthrough
+		case metTaintArg && isTaint:
+			taintArgs = append(taintArgs, s)
+		case !metTaintArg && !isTaint:
+			o.resources = append(o.resources, s)
+		case metTaintArg && !isTaint:
+			return fmt.Errorf("all resources must be specified before taint changes: %s", s)
+		}
+	}
+
+
+	if len(o.resources) < 1 {
+		return fmt.Errorf("one or more resources must be specified as <resource> <name> or <resource>/<name>")
+	}
+	if len(taintArgs) < 1 {
+		return fmt.Errorf("at least one taint update is required")
+	}
+
+	if o.taintsToAdd, o.removeTaintKeys, err = parseTaints(taintArgs); err != nil {
+		return cmdutil.UsageError(cmd, err.Error())
+	}
+
+	mapper, typer := f.Object(cmdutil.GetIncludeThirdPartyAPIs(cmd))
+	o.builder = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true)).
+		ContinueOnError().
+		NamespaceParam(namespace).DefaultNamespace().
+		SelectorParam(o.selector).
+		ResourceTypeOrNameArgs(o.all, o.resources...).
+		Flatten().
+		Latest()
+
+	o.f = f
+	o.out = out
+	o.cmd = cmd
+
+	return nil
+}
+
+// Validate checks to the TaintOptions to see if there is sufficient information run the command.
+func (o TaintOptions) Validate(args []string) error {
+	if !strings.HasPrefix(strings.ToLower(o.resources[0]), "node") {
+		return fmt.Errorf("invalid resource type %s, only nodes supported", o.resources[0])
+	}
+
+	// check the format of taint args and checks removed taints aren't in the new taints list
+	conflictKeys := []string{}
+	removeTaintKeysSet := sets.NewString(o.removeTaintKeys...)
+	for _, taint := range o.taintsToAdd {
+		if removeTaintKeysSet.Has(taint.Key) {
+			conflictKeys = append(conflictKeys, taint.Key)
+		}
+	}
+	if len(conflictKeys) > 0 {
+		return fmt.Errorf("can not both modify and remove the following taint(s) in the same command: %s", strings.Join(conflictKeys, ", "))
+	}
+
+	return nil
+}
+
+// RunTaint does the work
+func (o TaintOptions) RunTaint() error {
+	r := o.builder.Do()
+	if err := r.Err(); err != nil {
+		return err
+	}
+
+	return r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		obj, err := info.Mapping.ConvertToVersion(info.Object, info.Mapping.GroupVersionKind.GroupVersion().String())
+		if err != nil {
+			return err
+		}
+		name, namespace := info.Name, info.Namespace
```

Comment:
The name here is the node name, the namespace is empty string since nodes don't have namespaces.
I debugged this part, it looks working fine.